src/os/bluestore/Btree2Allocator.h

```diff
@@ -0,0 +1,425 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <mutex>
+
+#include "include/cpp-btree/btree_map.h"
+#include "include/cpp-btree/btree_set.h"
+
+#include "Allocator.h"
+
+#include "os/bluestore/bluestore_types.h"
+#include "include/mempool.h"
+#include "common/ceph_mutex.h"
+
+template <size_t BASE_BITS, size_t NUM_BUCKETS, size_t MUX = 1>
+struct SizeBucketCollectionBase {
```

Comment:
I think typical nomenclature for classes like this is *Traits.
It does not provide any meaningful base behavior, only settings.

Reply:
Renamed to ExtentCollectionTraits and moved to Allocator.h